go/example_code/s3/delete-buckets-by-prefix/s3_delete_buckets.go

```diff
@@ -0,0 +1,105 @@
+/*
+   Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+   This file is licensed under the Apache License, Version 2.0 (the "License").
+   You may not use this file except in compliance with the License. A copy of
+   the License is located at
+
+    http://aws.amazon.com/apache2.0/
+
+   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations under the License.
+*/
+package main
+
+import (
+    "flag"
+    "fmt"
+    "os"
+    "strings"
+
+    "github.com/aws/aws-sdk-go/aws"
+    "github.com/aws/aws-sdk-go/aws/session"
+    "github.com/aws/aws-sdk-go/service/s3"
+    "github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+func deleteBucketsByPrefix(sess *session.Session, prefix string) error {
+    // Create Amazon S3 service client
+    svc := s3.New(sess)
+
+    // Empty list to hold names of S3 buckets with prefix
+    bucketList := make([]string, 0)
+
+    result, err := svc.ListBuckets(nil)
+    if err != nil {
+        fmt.Println("Could not list buckets")
+        os.Exit(1)
```

Comment:
return the error so you give the user code an opportunity to react to it

Reply:
Good point.